### PR TITLE
narrowlink: init at 0.1.4

### PR DESCRIPTION
**Pull request by dit7ya**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/248617.

Disclaimer: I haven't tested the resulting binaries thoroughly yet.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/narrowlink/default.nix
Comment (alyssais) [inline pkgs/tools/networking/narrowlink/default.nix:12]: 0.1.3 is out, but it's fine to leave that for follow-up IMO.

File: pkgs/tools/networking/narrowlink/default.nix
Comment (alyssais) [inline pkgs/tools/networking/narrowlink/default.nix:26]: Would be nice if upstream kept this in version control.  Maybe ask them?

**Review by alyssais — CHANGES_REQUESTED** [review]
Could you please mark this broken on x86_64-darwin (or fix it!)?

Otherwise, LGTM. :)

File: pkgs/tools/networking/narrowlink/default.nix
Comment (SajjadPourali) [reply pkgs/tools/networking/narrowlink/default.nix:26]: Done - https://github.com/narrowlink/narrowlink/commit/37def5c5cfd62a4d623d92fdf03d4e8d3eac0e08

Comment (dit7ya) [reply pkgs/tools/networking/narrowlink/default.nix:26]: @SajjadPourali can you please cut a release after the lock-file addition commit? That would make automating updating this package a bit simpler here. 

Also since you commented, do you have any idea why the build is failing on `x86_64-darwin` - build logs here - https://logs.ofborg.org/?key=nixos/nixpkgs.248723&attempt_id=d7ae30ff-d5b7-4cb0-9d55-884033c82973

Comment (SajjadPourali) [reply pkgs/tools/networking/narrowlink/default.nix:26]: I think the failure is due to using Apple SDK 10 instead of 11. Please take a look at this issue for more info: https://github.com/GuillaumeGomez/sysinfo/issues/915

------

For the tag/release, does https://github.com/narrowlink/narrowlink/releases/tag/Nightly works for you? It might take a bit time for the actual release.

Comment (dit7ya) [reply pkgs/tools/networking/narrowlink/default.nix:26]: Thanks! The build failure is solved! 

Using the `Nightly` tag will imply making the version `unstable-DDDD-MM-YY`. Do you think cutting a minor or patch release is possible?

There are workarounds

Comment (SajjadPourali) [reply pkgs/tools/networking/narrowlink/default.nix:26]: The new release will be done after https://github.com/narrowlink/narrowlink/pull/28. I plan to finish it by this weekend.

**Comment by JeremiahSecrist** [comment]
Once this gets merged I'll make a nixos module for it.

File: pkgs/tools/networking/narrowlink/default.nix
Comment (SajjadPourali) [reply pkgs/tools/networking/narrowlink/default.nix:26]: > Thanks! The build failure is solved!
> 
> Using the `Nightly` tag will imply making the version `unstable-DDDD-MM-YY`. Do you think cutting a minor or patch release is possible?
> 
> There are workarounds but not clean.

@dit7ya Done

Comment (dit7ya) [reply pkgs/tools/networking/narrowlink/default.nix]: Done, the Cargo.lock is still outdated but I figured using a patch (much smaller than checking in the entire lockfile) is faster than waiting for the next release.

**Review by alyssais — APPROVED** [review]

File: pkgs/tools/networking/narrowlink/default.nix
Comment (dit7ya) [inline pkgs/tools/networking/narrowlink/default.nix:38]: Damn, I forgot to remove the package name from the description. I guess its for the next update then :shrug: .

**Pull request merged**
